api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -21,6 +22,11 @@ def on_connect(event, context)
   # Add in some logging to make debugging easier
   puts request_context
 
+  # Return early if this is the user connectivity test
+  authorizer = request_context['authorizer']
+  if authorizer && authorizer['connectivityTest']
+    return { statusCode: 200, body: "connection successful" }
```

Comment:
My knowledge of websockets is limited. Does this return an HTTP 200 response to the initial HTTP Upgrade request? Can we also send a message on the websocket to the browser and require the client to wait for that message to validate that bi-directional messaging is working? I’m guessing that the `$connect` route can’t do that.

Reply:
Take a look at the `default` route (below) and the client-side part of the test: https://github.com/code-dot-org/code-dot-org/pull/40250

With those two, we make sure the bi-directional message also happens.

Reply:
Never mind. I see that the default route supports a bidirectional exchange of messages!